src/lib/phdi-building-blocks/phdi_building_blocks/fhir.py

```diff
@@ -4,83 +4,54 @@
 from urllib3 import Retry
 
-from azure.core.credentials import AccessToken
 from azure.identity import DefaultAzureCredential
 from azure.storage.blob import download_blob_from_url
+from phdi_building_blocks.classes.creds_manager import AzureFhirserverCredentialManager
 
 
 def generate_filename(blob_name: str, message_index: int) -> str:
-    """Strip the file type suffix from the blob name, and append message index."""
-    fname = blob_name.split("/")[-1]
-    fname, _ = fname.rsplit(".", 1)
-    return f"{fname}-{message_index}"
-
-
-class AzureFhirserverCredentialManager:
-    """Manager for handling Azure credentials for access to the FHIR server"""
-
-    def __init__(self, fhir_url):
-        """Credential manager constructor"""
-        self.access_token = None
-        self.fhir_url = fhir_url
-
-    def get_fhir_url(self):
-        """Get FHIR URL"""
-        return self.fhir_url
-
-    def get_access_token(self, token_reuse_tolerance: float = 10.0) -> AccessToken:
-        """If the token is already set for this object and is not about to expire
-        (within token_reuse_tolerance parameter), then return the existing token.
-        Otherwise, request a new one.
-        :param str token_reuse_tolerance: Number of seconds before expiration
-        it is OK to reuse the currently assigned token"""
-        if not self._need_new_token(token_reuse_tolerance):
-            return self.access_token
-
-        creds = self._get_azure_credentials()
-        scope = f"{self.fhir_url}/.default"
-        self.access_token = creds.get_token(scope)
-
-        return self.access_token
-
-    def _get_azure_credentials(self):
-        """Get default Azure Credentials from login context and related
-        Azure configuration."""
-        return DefaultAzureCredential()
-
-    def _need_new_token(self, token_reuse_tolerance: float = 10.0) -> bool:
-        """Determine whether the token already stored for this object can be reused, or if it
-        needs to be re-requested.
-        :param str token_reuse_tolerance: Number of seconds before expiration
-        it is OK to reuse the currently assigned token"""
-        try:
-            current_time_utc = datetime.now(timezone.utc).timestamp()
-            return (
-                self.access_token.expires_on - token_reuse_tolerance
-            ) < current_time_utc
-        except AttributeError:
-            # access_token not set
-            return True
-
-
-def get_fhirserver_cred_manager(fhir_url: str):
-    """Get an instance of the Azure FHIR Server credential manager."""
+    """
+    Strip the file type suffix from the blob name, and instead
+    append the message index.
+    :param str blob_name: The name of the blob to modify
+    :param int message_index: The index of this message in the batch
+    :return: The derived filename
+    :rtype: str
+    """
+    filename = blob_name.split("/")[-1]
+    root_name, _ = filename.rsplit(".", 1)
```

Comment:
I have tripped over this many times - in various tools and libraries I've used, sometimes filename includes extension, and other times not.  Naming for files is all over the place.

Straw man to standardize on:
`file_location` = /mydirectory/myfile.ext
`file_path` = /mydirectory/
`filename` = myfile
`extension` = ext
`full_filename` = myfile.ext

In this case we have a before and after filename, so maybe we could rename `root_name` to `original_filename`.

Reply:
I really like this. But your last sentence doesn't seem to vibe with your recommendation. Wouldn't we change `filename` to be `full_filename`, `root_name` to `filename`, and `_` to `ext`? Or am I misunderstanding?

Reply:
I like the idea of agreeing on a standard file and path names. @spencer-skylight is spot on that this can be all over the place. I don't have strong feelings on the exact names we choose, but would like to ask that whatever the decision is be recorded in some kind of phdi_build_blocks style guide. I know we aren't looking to flesh out a whole guide currently, but what about a place holder .md file with one section on file and path variable naming? I'd hate for this to get lost in one conversation buried in a reasonably large closed and merged PR.

Reply:
I think we should definitely put this in both a style guide (eventually) and potentially in an ADR in the mean time. That could actually go along well with the ADR around versioning and the adoption of SemVer. In this case, root was intended to be the file without the extension, which it looks like was grokked correctly, but I agree the naming for these kinds of things is terrible. I don't like how the proposed system uses `file_path` to refer to the parent directory when to me, a file path intuitively refers to the actual full path to the file (i.e. corresponds to file_location), but that's getting into semantics. For now, I'll change the variables to be super descriptive versions of themselves until we figure out what semantics we'd like to go with.

Reply:
I realize this is huge PR that has been in review for a long time, and I really don't want to cause further delay, but I'm going to push back on punting setting a standard around file names and paths. To my mind this is exactly the sort of thing that should come out of this PR in order to prevent the need for future style-motivated refactors. In PR 129, that I'm getting ready to merge, does a bunch of file I/O. Several times during development  I came back to this conversation to see if it had resulted in a decision so that I could adopt it and prevent the need for future refactoring. The longer we go without an answer around this the more variable names we will have to clean up later. To this end what do you y'all this of the following:

full_file_path = /mydirectory/myfile.ext
parent_directory_path = /mydirectory/
filename = myfile
extension = ext
full_extension = .ext
full_filename = myfile.ext

Reply:
+1 to not punting. This is definitely the type of stuff that should come out of this PR.

Given the two proposed taxonomy's, I'd prefer we go with Spencer's. I think it's simpler in its naming, and I can't foresee a scenario where needing to classify something as the `parent_path` compared to just the `file_path` would be a necessary distinction. I also don't think we need to differentiate between the full extension, and the non-full extension. Instead, we should just always assume that the extension does _not_ include the period.

Reply:
Personally, I'm perfectly happy with Spencer's nomenclature. I was just trying to address @bamader's comment about `file_path`, to move towards consensus. I included `extension` and `full_extension` because disambiguating between with and without the period isn't something that had occurred to me before. Provided we agree to always go without the period I don't see much value in it.

Reply:
The consensus is to go with Spencer's recommendation, which has been added to a draft version of a Design Guide. Thanks for the great back and forth here!